pymc/logprob/rewriting.py

```diff
@@ -351,3 +351,18 @@ def construct_ir_fgraph(
         fgraph.replace_all(new_to_old)
 
     return fgraph, rv_values, memo
+
+
+pregrad_logprob_rewrites_db = SequenceDB()
+pregrad_logprob_rewrites_db.name = "pregrad_logprob_rewrites_db"
+pregrad_logprob_rewrites_db.register("pre-canonicalize", optdb.query("+canonicalize"), "basic")
+pregrad_logprob_rewrites_db.register("stabilize", optdb.query("+stabilize"), "basic")
```

Comment:
I wonder if this might be an overkill, should we start with only cherry picked rewrites?

Reply:
@ricardoV94 I'm not sure what we should do here. What kind of inclusion criteria are you thinking? I imagine anything that simplifies or improves the stability of the logp graph would be nice?

Reply:
I was thinking of starting with just the things we know about. The reason for this is that we might want to use this as the "base" for pre-grad rewrites in PyTensor if/when we decide to make them lazy. This would be a good way to start learning what rewrites seem useful for gradients...  Probably everything in `canonicalize` makes sense as that makes the graphs "homogenous", but everything in `stabilize` might be an overkill or even counterproductive. `stabilizing` the base graph may make the gradient one more complex.

On another angle, I worry that the extra rewrites could make compilation much slower.

Could you perhaps benchmark how long does it take to obtain `dlogp` and, separately, to compile it?
The compilation should be checked with and without having already cached results. Youu can erase the pytensor cache with `pytensor-cache purge` or something similar.

Reply:
Yeah, I think this would be the easiest. If the rewrite time is not too affected by including all of `stabilize` vs not (just to test more easily), then I am okay with the current approach.